src/tasks/AppleAppBuilder/Templates/runtime-coreclr.m

```diff
@@ -0,0 +1,161 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#import <Foundation/Foundation.h>
+#include "coreclrhost.h"
+#include <TargetConditionals.h>
+#import <os/log.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <dlfcn.h>
+
+#import "util.h"
+
+#define APPLE_RUNTIME_IDENTIFIER "iossimulator-arm64"
+
+const char *
+get_bundle_path (void)
+{
+    static char *bundle_path = NULL;
+    if (bundle_path)
+        return bundle_path;
+    NSBundle* main_bundle = [NSBundle mainBundle];
+    NSString* path = [main_bundle bundlePath];
+
+#if TARGET_OS_MACCATALYST
+    path = [path stringByAppendingString:@"/Contents/Resources"];
+#endif
+
+    bundle_path = strdup ([path UTF8String]);
+
+    return bundle_path;
+}
+
+char *
+compute_trusted_platform_assemblies ()
+{
+    const char *bundle_path = get_bundle_path ();
+
+    NSMutableArray<NSString *> *files = [NSMutableArray array];
+    NSMutableArray<NSString *> *exes = [NSMutableArray array];
+
+    NSFileManager *manager = [NSFileManager defaultManager];
+    NSString *dir = [NSString stringWithUTF8String: bundle_path];
+    NSDirectoryEnumerator *enumerator = [manager enumeratorAtURL:[NSURL fileURLWithPath: dir]
+                                                 includingPropertiesForKeys:@[NSURLNameKey, NSURLIsDirectoryKey]
+                                                 options:NSDirectoryEnumerationSkipsSubdirectoryDescendants
+                                                 errorHandler:nil];
+    for (NSURL *file in enumerator) {
+        // skip subdirectories
+        NSNumber *isDirectory = nil;
+        if (![file getResourceValue:&isDirectory forKey:NSURLIsDirectoryKey error:nil] || [isDirectory boolValue])
+            continue;
+
+        NSString *name = nil;
+        if (![file getResourceValue:&name forKey:NSURLNameKey error:nil])
+            continue;
+        if ([name length] < 4)
+            continue;
+       if ([name compare: @".dll" options: NSCaseInsensitiveSearch range: NSMakeRange ([name length] - 4, 4)] == NSOrderedSame) {
+            [files addObject: [dir stringByAppendingPathComponent: name]];
+        }
+    }
+
+    // Join them all together with a colon separating them
+    NSString *joined = [files componentsJoinedByString: @":"];
+    return strdup([joined UTF8String]);
+}
+
+void*
+pinvoke_override (const char *libraryName, const char *entrypointName)
+{
+    if (!strcmp (libraryName, "__Internal"))
+    {
+        return dlsym (RTLD_DEFAULT, entrypointName);
+    }
+    return NULL;
+}
+
+void
+mono_ios_runtime_init (void)
```

Comment:
Nit: maybe once this is merged, we should followup with some naming cleanup in the samples and rename this to just `ios_runtime_init`...